lib/astarte_data_access/data.ex

```diff
@@ -22,121 +22,143 @@ defmodule Astarte.DataAccess.Data do
       AND endpoint_id=:endpoint_id AND path=:path
     """
 
-    fetch_property_query =
-      Query.new()
-      |> Query.statement(fetch_property_value_statement)
-      |> Query.put(:device_id, device_id)
-      |> Query.put(:interface_id, interface_descriptor.interface_id)
-      |> Query.put(:endpoint_id, mapping.endpoint_id)
-      |> Query.put(:path, path)
-      |> Query.consistency(:quorum)
-
-    with {:ok, result} <- Query.call(db_client, fetch_property_query),
-         [{_column, value}] when not is_nil(value) <- Result.head(result) do
-      {:ok, value}
-    else
-      :empty_dataset ->
-        {:error, :property_not_set}
+    fetch_property_value_params = %{
+      device_id: device_id,
+      interface_id: interface_descriptor.interface_id,
+      endpoint_id: mapping.endpoint_id,
+      path: path
+    }
 
-      [{column, nil}] when is_atom(column) ->
-        Logger.warn("Unexpected null value on #{path}, mapping: #{inspect(mapping)}.")
-        {:error, :undefined_property}
+    with {:ok, prepared} <- Xandra.prepare(conn, fetch_property_value_query),
+         {:ok, %Xandra.Page{} = page} <-
+           Utils.retrieve_page(conn, prepared, fetch_property_value_params, consistency: :quorum) do
+      retrieve_property_value(page)
+    end
+  end
 
-      %{acc: _, msg: error_message} ->
-        Logger.warn("fetch_property: database error: #{error_message}")
-        {:error, :database_error}
+  defp retrieve_property_value(%Xandra.Page{} = page) do
+    case Enum.to_list(page) do
+      [] ->
+        {:error, :property_not_set}
+
+      [property_map] ->
+        # The map key can be any of double_value, integer_value, etc...
+        [value] = Map.values(property_map)
 
-      {:error, reason} ->
-        Logger.warn("Database error while retrieving property: #{inspect(reason)}")
-        {:error, :database_error}
+        if value != nil do
+          {:ok, value}
+        else
+          {:error, :undefined_property}
+        end
     end
   end
 
   @spec path_exists?(
-          :cqerl.client(),
+          String.t(),
           Device.device_id(),
           %InterfaceDescriptor{},
           %Mapping{},
           String.t()
         ) :: {:ok, boolean} | {:error, atom}
-  def path_exists?(db_client, device_id, interface_descriptor, %Mapping{} = mapping, path)
+  def path_exists?(
+        realm,
+        device_id,
+        interface_descriptor,
+        mapping,
+        path
+      )
       when is_binary(device_id) and is_binary(path) do
+    with true <- Utils.verify_realm_ok(realm) do
+      Xandra.Cluster.run(
+        :astarte_data_access_xandra,
+        &do_path_exists?(&1, realm, device_id, interface_descriptor, mapping, path)
+      )
+    end
+  end
+
+  defp do_path_exists?(conn, realm_name, device_id, interface_descriptor, mapping, path) do
     # TODO: do not hardcode individual_properties here
-    fetch_property_value_statement = """
+    fetch_property_value_query = """
```

Comment:
Btw I'm not sure why you changed all the `query` to `statement` since the documentation for [`prepare`](https://hexdocs.pm/xandra/Xandra.html#prepare/3) actually calls that argument "statement"